### PR TITLE
[inductor] Preserve dense size-1 strides for *_like under preserve_format

### LINKED ISSUE #197815: `torch.compile` (inductor) — `*_like` ops don't preserve a size-1 dimension's stride under `preserve_format` (eager does)
### 🐛 Describe the bug

The `*_like` factory ops (`ones_like`, `zeros_like`, `empty_like`, `rand_like`, `full_like`)
default to `memory_format=torch.preserve_format`, i.e. the output should keep the input tensor's
strides. Under `torch.compile` (inductor), when the input has a **size-1 dimension** with a
non-standard stride, the compiled output does not preserve that dimension's stride (eager does),
so eager and compiled disagree on the output's strides. The values are correct; only the stride
metadata differs.

## Minimal repro

```python
import torch

x = torch.randn(1, 5, device='cuda').t()   # shape (5, 1), stride (1, 5); 5 elements

def fn(t):
    return torch.ones_like(t)               # default memory_format=preserve_format

print(x.stride())                                        # (1, 5)
print(fn(x).stride())                                    # eager:    (1, 5)  -- size-1 dim stride preserved
print(torch.compile(fn, backend="inductor")(x).stride()) # compiled: (1, 1)  -- size-1 dim stride changed
```

The whole `*_like` family is affected, always on a **size-1** dimension:

| input | numel | eager stride | compiled stride |
|---|---|---|---|
| `randn(1,5).t()` → (5,1) | 5 | (1, 5) | (1, **1**) |
| `randn(2,4,1).transpose(1,2)` → (2,1,4) | 8 | (4, 1, 1) | (4, **4**, 1) |
| `randn(4,1,5).permute(1,0,2)` → (1,4,5) | 20 | (5, 5, 1) | (**20**, 5, 1) |
| `randn(3,1,4).transpose(0,1)` → (1,3,4) | 12 | (4, 4, 1) | (**12**, 4, 1) |

Genuine permutations with **no size-1 dim** (`randn(3,4).t()`, a channels-last 4-D tensor,
`permute(2,0,1)`) are preserved correctly by both, so the gap is specific to the size-1 dimension.

## Expected behavior

With `memory_format=preserve_format` (the default), the compiled output should keep the input's
strides — including the stride of a size-1 dimension — matching eager.

## Actual behavior

Eager preserves the size-1 dimension's stride (e.g. `(1, 5)`); inductor produces a normalized
stride (e.g. `(1, 1)`). Values are identical; the output strides differ. The differing stride is
always on a size-1 dimension, so element addressing is unaffected — this is a metadata-only
divergence, but it is still a `preserve_format` mismatch between eager and compiled.

### Versions

```
PyTorch version: 2.15.0a0+gite67e43b
Is debug build: True
CUDA used to build PyTorch: 12.6
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.4 LTS (x86_64)
GCC version: (Ubuntu 12.3.0-1ubuntu1~22.04.3) 12.3.0
Clang version: 17.0.6
CMake version: version 4.0.0
Libc version: glibc-2.35

Python version: 3.12.13 | packaged by conda-forge (64-bit runtime)
Python platform: Linux-6.8.0-138-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 12.6.20
GPU 0: NVIDIA RTX A6000
Nvidia driver version: 580.82.07
cuDNN version: 9.3.0

[pip3] numpy==2.3.5
[pip3] torch==2.15.0a0+gite67e43b
[pip3] triton==3.7.1
```

cc @jamesr66a @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

### PR DESCRIPTION
**Pull request by indrajeetapache**
## Summary

Fixes #197815.

With default `memory_format=torch.preserve_format`, eager `*_like` keeps the input strides — including non-canonical strides on **size-1** dims. Inductor did not:

```python
x = torch.randn(1, 5).t()  # shape (5, 1), stride (1, 5)
torch.ones_like(x).stride()                          # eager: (1, 5)
torch.compile(lambda t: torch.ones_like(t))(x).stride()  # was: (1, 1)
```

Values matched; only stride metadata diverged. That breaks anything that compares strides (e.g. HOP / while_loop metadata).

**Root cause:** refs `empty_like` and Inductor `full_like` / `_rand_like` rebuild layout via `compute_elementwise_output_logical_to_physical_perm`. Contiguity skips size-1 dims, so dense tensors with unusual size-1 strides look contiguous and get **canonical** strides. Eager ATen uses `empty_strided(sizes, strides)` for non-overlapping dense inputs.

**Fix:** when `is_non_overlapping_and_dense_or_false`:

1. `torch/_refs/__init__.py` `empty_like` → `empty_strided(shape, stride)` (covers refs `ones_like` / `zeros_like` / etc. that go through it)
2. Inductor `full_like` / `_rand_like` → allocate with exact strides via `empty_strided` + `copy_` when the filled/random contiguous buffer differs

Same idea as the dense short-circuit in closed #185304, scoped to the `*_like` preserve_format path.

## Test plan

- [x] `test/inductor/test_torchinductor.py::test_like_preserves_dense_size_one_strides`
- [x] `test/test_fake_tensor.py::test_like_preserve_format_preserves_dense_strides`
- [ ] CI: inductor / fake tensor

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @jansel @jamesr66a

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes
